packages/_helpers/decorators.ts

```diff
@@ -0,0 +1,207 @@
+import { h, Component, ComponentProps, PropOptions, prop as skProp, emit, define } from 'skatejs';
+import { ColorType } from './colorTypes';
+import { IS_DEV } from './environment';
+import { scopeCss, Constructable } from './utils';
+
+export function renderCss(): MethodDecorator {
+  return function <T extends Function>(
+    target: Object,
+    propertyKey: string | symbol,
+    descriptor: TypedPropertyDescriptor<T>,
+  ) {
+    return {
+      value( this: { css?: string, shadyCss?: string | void } & Object ) {
+        // we preffer shadyCss which is present if Css Mixin is used, otherwise fallback to css property
+        // tslint:disable-next-line:no-invalid-this
+        const css = 'shadyCss' in this ? this.shadyCss : this.css;
+        // tslint:disable-next-line:no-invalid-this
+        return [ h( 'style', css ) ].concat( descriptor.value.call( this, this ) );
+      }
+    };
+  };
+}
+
+export function customElement( name: string ): ClassDecorator {
+  return function <T extends Constructable<Component<any>>>( Target: T ): T {
+    Object.defineProperty( Target, 'is', {
+      configurable: true,
+      get() { return name; }
+    } );
+    return define( Target );
+  };
+}
+
+type PropConfig = PropOptions<any, any> & { type?: Function };
+type PropType = 'string' | 'number' | 'object' | 'array' | 'boolean';
+const identityFn = <T>( _: T ) => _;
+
+export function prop( config: PropConfig = {} ): PropertyDecorator {
+  return function( targetProto: { [ key: string ]: any } & Component<any>, propertyKey: string | symbol ) {
+    const currentValue = targetProto[ propertyKey ];
+    const { type, ...skPropConfig } = config;
+    const configType = parseType( type );
+    const skatePropTypeFn = skProp[ configType ] || identityFn;
+    const Ctor = targetProto.constructor as typeof Component;
+    const existingProps = ( Ctor.props || {} ) as ComponentProps<any, any>;
+    const newProps = {
+      ...existingProps,
+      ...{ [ propertyKey ]: skatePropTypeFn( skPropConfig ) }
+    };
+    Object.defineProperty(
+      Ctor,
+      'props',
+      {
+        configurable: true,
+        get() { return newProps; }
+      }
+    );
+
+    // this is needed for Babel ... :-/
```

Comment:
Babel is using obsolete version of decorator proposal. Even default values doesn't work... One of many reasons why I prefer TS as transpiler